Adding at risk features to charter exception on class 4 changes

iherman opened the pull request:
This PR tries to answer to issue #114 raised by @jandrieu. 

TL;DR: beyond the cases of security issues, the scope section adds one more exception to the rule disallowing class 4 changes: properties that are (currently) "at risk" and that may have to be removed from the final spec because it "they did not make it". In other words, if the community/WG finds a consensus for, say, the `refreshService` term, and then it passes the CR Criteria, then it can be added to the spec during the lifetime of the WG.

I listed all the terms that are included in the VCDM spec right now. Please check if I forgot any of those.

(I prefer to list the terms explicitly. The AC, sometimes, frowns upon open-ended charters, and I wanted to avoid that.)


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-wg-charter/pull/119.html" title="Last updated on Jun 5, 2024, 4:41 AM UTC (64c4884)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-wg-charter/119/8eef289...64c4884.html" title="Last updated on Jun 5, 2024, 4:41 AM UTC (64c4884)">Diff</a>

iherman committed:
Adding the at risk features

iherman committed:
Added linebreaks

iherman committed:
Misplaced ; character

jandrieu commented on index.html:
```diff
@@ -171,8 +171,27 @@ <h2>Scope</h2>
+
+        <ul>
+          <li>
+            For features that had been listed as "At Risk" during the CR phase of the Verifiable Data Model 
+            specification, had been removed from the final document due to missing the CR Exit Criteria, but 
+            have been developed further by the community providing a possibly adequate consensus and support 
+            to be added to the Recommendation. At the time of writing this charter, these terms are: 
```
```suggestion
            For features that had been listed as "At Risk" during the CR phase of the Verifiable Data Model 
            specification, had been removed from the final document due to missing the CR Exit Criteria, but 
            have been developed further by the community providing a possibly adequate consensus and support 
            to be added to the Recommendation. At the time of writing this charter, these terms are: 
```

This section seems to be missing the predicate that says that "such features shall be in scope" or however the language might go.

jandrieu replied:
I see it now.

Confusion resolved.

TallTed commented on index.html:
```diff
@@ -171,8 +171,27 @@ <h2>Scope</h2>
+
+        <ul>
+          <li>
+            For features that had been listed as "At Risk" during the CR phase of the Verifiable Data Model 
```
```suggestion
            Features that had been listed as "At Risk" during the CR phase of the Verifiable Data Model 
```

TallTed commented on index.html:
```diff
@@ -171,8 +171,27 @@ <h2>Scope</h2>
+            <code>RefreshService</code>, <code>ConfidenceMethod</code>, etc.).
+          </li>
+          <li>
+            If serious security issues come to the fore that require changes in a Recommendation.
```
```suggestion
            Serious security issues that arise, requiring changes in a Recommendation.
```

msporny reviewed (approved):
As long as nits/clarifications by others that are included, LGTM.

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-05-29-vcwg#section2-1) on 2024-05-29

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.1. Adding at risk features to charter exception on class 4 changes (pr vc-wg-charter#119)

_See github pull request [vc-wg-charter#119](https://github.com/w3c/vc-wg-charter/pull/119)._




**Brent Zundel:** this PR is in response to request for changes from the group.  
… it clarifies the scope of the group.  
… happy to take comments from the queue.  

**Ivan Herman:** it doesn't clarify. it adds something to the scope.  
… previously we were not able to make class 4 changes.  
… we did have one exception--serious security problems.  
… so I have added another exception, that any of the features that are at risk in the VCDM, any of those can be picked up and added to the recommendation--if all goes well.  
… I think that handles JoeAndrieu's concerns.  
… to the group: please confirm that is the right list in the PR.  
… and make sure I didn't forget anything.  
… I want to make it an explicit list, as the AC doesn't like open ended charters.  

**Joe Andrieu:** thank you, ivan.  
… one question, this would include making new standards track docs to deal with these standards track documents.  
… I'd like the ability to create those specs during the 2 year charter.  

**Ivan Herman:** essentially that's correct.  
… we've cut documents in two before, for example.  
… it's not explicit about how things are taken out or put in practically.  
… so if it's an addition to a spec or a new spec is not explicitly stated.  
… so the answer to your question is "yes" that is possible to make new specs with those features.  

> *Joe Andrieu:* Great. thanks for the clarification.

**Brent Zundel:** that's my understanding also.  

**Manu Sporny:** and mine also.  
… now to poke the charter bear a bit...  
… if I understand this correctly, we want to publish VCDM 2.0 sooner than later.  
… but our charter would let us add the at risk features to a v2.1 edition.  
… which could put the features back.  

**Ivan Herman:** yes.  

**Manu Sporny:** from an editorial perspective, some of these features may be cut due to lack of implementation.  
… but starting in January, those could get added back with proper implementations and tests.  

**Ivan Herman:** correct.  
… for those not familiar with "class 4" these have to be existing features stated as "at risk" in the documents.  
… oh, and one more thing, the current charter which we are under now was extended to the end of the year.  

**Brent Zundel:** great. are there any objections.  
… please review the PR and comment there.  
… it was opened this morning, so by our next meeting we will merge it assuming all are in agreement.  
… and then we can move the draft charter forward for review.  

**Manu Sporny:** what's the timeline?  

**Ivan Herman:** ASAP.  

**Manu Sporny:** this goes to W3M, then the AC, then the vote starts in July?  

**Ivan Herman:** that sounds realistic...maybe a bit earlier.  
… the changes are explicitly very small.  
… the paragraph in the PR is the only meaningful change.  
… the horizontal review folks did not have anything to add.  
… with the AC, though, nobody knows.  
… but assuming no objections, this new charter will take over in January.  

</details>

iherman committed:
Apply Ted's suggestions from code review

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

iherman committed:
Updated the reference charter and regenerated the diff file for review.
